subsys/dfu/dfu_target/src/dfu_target_suit.c

```diff
@@ -65,88 +62,124 @@ int dfu_target_suit_init(size_t file_size, int img_num, dfu_target_callback_t cb
 		return -ENODEV;
 	}
 
-	if (file_size > DFU_PARTITION_SIZE) {
-		LOG_ERR("Requested file too big to fit in flash %zu > 0x%x", file_size,
-			DFU_PARTITION_SIZE);
+	if (ENVELOPE_IMAGE_NUMBER == img_num) {
+		/* Get info about dfu_partition to store the envelope */
+		err = suit_dfu_partition_device_info_get(&device_info);
+	} else {
+		/* Cache partitions ids starts from 0, whereas image number 0 is reserved for
+		 * dfu_partition. Decrease img_num value by 1 to reach the correct DFU cache
+		 * partition id.
+		 */
+#ifdef CONFIG_SUIT_CACHE_RW
+		err = suit_dfu_cache_rw_device_info_get(img_num - 1, &device_info);
+#endif
```

Comment:
Should not it fail if `CONFIG_SUIT_CACHE_RW` is not defined and `ENVELOPE_IMAGE_NUMBER != img_num`?
Seems like uninitialized variable usage in line 81.

Reply:
Fixed.